src/epr/client.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+# SPDX-FileCopyrightText: © 2024 Brett Smith <xbcsmith@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import logging
+from typing import Any, Optional
+from urllib.parse import urljoin
+
+import urllib3
+
+from .common import EnhancedJSONEncoder
+
+urllib3.disable_warnings()
+
+logger = logging.getLogger(__name__)
+
+
+class Client(object):
+    def __init__(self, url, headers=None):
+        self.url = url
+        self.api_version = "v1"
+        self.graphql_query = "graphql/query"
+        if self.url is None:
+            self.url = "http://localhost:8042"
+        self.endpoint = "/".join(["api", self.api_version, self.graphql_query])
+        self.target = urljoin(self.url, self.endpoint)
+        self.headers = {"Content-Type": "application/json"}
+        if headers is not None:
+            self.headers.update(headers)
+        self._operation_map = {
+            "search": {
+                "events": "FindEventInput!",
+                "event_receivers": "FindEventReceiverInput!",
+                "event_receiver_groups": "FindEventReceiverGroupInput!",
+            },
+            "mutation": {
+                "create_event": "CreateEventInput!",
+                "create_event_receiver": "CreateEventReceiverInput!",
+                "create_event_receiver_group": "CreateEventReceiverGroupInput!",
+            },
+            "operation": {
+                "events": "event",
+                "event_receivers": "event_receiver",
+                "event_receiver_groups": "event_receiver_group",
+            },
+            "create": {
+                "create_event": "event",
+                "create_event_receiver": "event_receiver",
+                "create_event_receiver_group": "event_receiver_group",
+            },
+        }
+
+    def new_graphql_search_query(
+        self, operation: str, params: Optional[dict] = None, fields: Optional[list] = None
+    ) -> dict:
```

Comment:
Suggest creating dataclass for graph queries so the return here is self documenting, e.g.,
```python
@dataclass
class GraphQLQuery:
    query: str
    variables: str
```
Same could be useful for `new_graphql_mutation_query()` as well.

Reply:
Good suggestion. Based on my intention that Dataclasses should not be required to use the library I might not implement this.

Reply:
I'm fine with this but could you please explain why you don't want dataclasses required for library use? Are you trying to minimize what's exposed in the lib, or is it something with dataclasses specifically?